### PR TITLE
refactor: Refactor storage api calls

### PR DESCRIPTION
**Pull request by MichaelGrinstead**
Resolves #

- [ ] There is an associated issue (**required**)
- [x] The change is described in detail
- [ ] There are new or updated tests validating the change (if applicable)

## Description

- Separate out Pinata logic from the API route into a separate interface for data uploads
- Refactor api calls to use Tanstack Query
- Create custom hooks to fetch and upload data

### CODE REVIEW

File: apps/dapp/src/app/api/storage/route.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    const file = formData.get('file') as Blob;
+    const name = formData.get('name') as string;
+
+    if (file instanceof Blob && typeof name === 'string') {
+      const buffer = Buffer.from(await file.arrayBuffer());
+
+      const readableStream = new Readable({
+        read() {
+          this.push(buffer);
+          this.push(null);
```
Comment (DoubleOTheven) [inline apps/dapp/src/app/api/storage/route.ts:22]: Why do you push null?

File: apps/dapp/src/app/api/storage/route.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        },
+      });
+
+      result = await uploadFile(readableStream, name);
+    } else {
+      result = { ok: false, error: FileError.InvalidDataError };
+    }
+  } else if (contentType?.includes('application/json')) {
+    const json = await request.json();
+    if (typeof json === 'object' && json !== null)
```
Comment (DoubleOTheven) [inline apps/dapp/src/app/api/storage/route.ts]: What happens if json is not an object? We should return an error

File: apps/dapp/src/components/Navbar/NavBar.tsx [hunk 3]
```diff
@@ -61,10 +61,10 @@ export const NavBar = ({
             href="/profile"
             className="flex gap-2 items-center justify-center"
           />
-        </div>
+        </div> */}
```
Comment (DoubleOTheven) [inline apps/dapp/src/components/Navbar/NavBar.tsx]: Do you intend to comment out this whole block? Let's just delete it if so

File: apps/dapp/src/components/Navbar/NavDropdown.tsx [hunk 4]
```diff
@@ -1,28 +1,43 @@
...
+      });
+    }
+  }, [getProfileDataError]);
 
   return (
     <DropdownMenu>
       <DropdownMenuTrigger className="focus:outline-none">
         <div className="flex gap-2 items-center">
-          {hasProfile && <UserProfileBadge name={name} imageUrl={image} />}
+          {hasProfile && profileData && (
```
Comment (DoubleOTheven) [inline apps/dapp/src/components/Navbar/NavDropdown.tsx]: is `hasProfile` necessary? `profileData` should be valid data or undefined. Less variables to deal with that way

File: apps/dapp/src/components/Profile/Edit/EditProfileForAccount.tsx [hunk 5]
```diff
@@ -75,6 +88,16 @@ export default function EditProfileForAccount({
     [isUpdateProfilePending, isUpdatingData],
   );
 
+  useEffect(() => {
```
Comment (DoubleOTheven) [inline apps/dapp/src/components/Profile/Edit/EditProfileForAccount.tsx]: NIT: I'm seeing this block of code a lot (line 68, 91, and other files too). We can move this to a new hook and do something like:

```tsx
export interface WithError {
  message: string;
}

export const useToastError = (withError: WithError | undefined, title?: string) => {
  const { toast } = useToast();
  
  useEffect(() => {
     toast({
        title: title || 'Error',
        description: withError.message,
        variant: 'error',
      });
    }, [withError, title]);
} 
```

Then use it like this:

```tsx
const { uploadProfileData, uploadProfileDataError } = useUploadProfileData();
useToastError(uploadProfileDataError, 'Upload Error');
```

File: apps/dapp/src/app/api/storage/route.ts (hunk 1 above)
Comment (MichaelGrinstead) [reply apps/dapp/src/app/api/storage/route.ts:22]: Pushing null signals the end of the stream

File: apps/dapp/src/components/Profile/Edit/EditProfileForAccount.tsx (hunk 5 above)
Comment (MichaelGrinstead) [reply apps/dapp/src/components/Profile/Edit/EditProfileForAccount.tsx]: Done

File: apps/dapp/src/components/Navbar/NavDropdown.tsx (hunk 4 above)
Comment (MichaelGrinstead) [reply apps/dapp/src/components/Navbar/NavDropdown.tsx]: removed

File: apps/dapp/src/app/api/storage/route.ts (hunk 2 above)
Comment (MichaelGrinstead) [reply apps/dapp/src/app/api/storage/route.ts]: done

**Pull request merged**